### DISCUSSION TITLE
Add confirmation popup's to critical UI interactions

### DISCUSSION THREAD

**Discussion by mha1** [EdgeTX/edgetx#1883] [Ideas]
Hi,

Just for the fun of it I tried to set up a model entirely using the radio and it's great touch user interface. In general it worked well but I fell repeatedly into some traps using selection lists. As the spacing between list entries is kinda small for grownup index fingers I sometimes hit the list entry next to  one I intended. The worst I experienced was hitting Reset instead of Edit on a channel in the outputs. Resetting min/max without further notice on a live model is a servo killer as min/max jumps immediately to +/- 100 (think of flaps) potentially forcing the servo into some mechanical end stops. I also remember creating new models instead of selecting a model or inserting new unwanted mix lines when trying to edit a existing one. 

My request is to add safety popup's to critical list entries like the one for deleting models:

![screenshot_tx16s_22-04-23_17-46-16](https://user-images.githubusercontent.com/5615068/164913911-bd0275f7-1d9d-4a40-8958-fdda21d6f080.png)

Here's a selection of potentially critical list entries that should not execute without confirmation. Create Model, Duplicate Model, Delete model (done), Reset, Copy sticks to subtrim, Copy trims to subtrim, Insert before/after. And I'm sure there may be others.

![screenshot_tx16s_22-04-23_17-46-50](https://user-images.githubusercontent.com/5615068/164914168-3f152296-f1b4-4e8c-b886-3ae5d3305557.png)
![screenshot_tx16s_22-04-23_17-49-48](https://user-images.githubusercontent.com/5615068/164914177-36ad7d8e-bfba-4d39-a876-bc0fbd942556.png)
![screenshot_tx16s_22-04-23_18-11-45](https://user-images.githubusercontent.com/5615068/164914182-a120fea9-fae3-4945-858d-6f0b3a09efdd.png)
![screenshot_tx16s_22-04-23_17-50-14](https://user-images.githubusercontent.com/5615068/164914188-c8a784d1-9309-4196-a110-f758deb7d250.png)

**Comment by gagarinlg** [comment]
I partially agree.
We are already working no switching the UI to a new library and there are ideas to rework the UI to make ir more moders and more touch friendly. The size of the ui input elements is even worse on NV14 and PL18, as those have higher DPI displays and the size of the input elements is essentially the same, in pixels.
My concern is the number of steps to do something. Every additional step is slowing sdown the work flow, so additional safety checks should only be added, where they are needed. Accidentally copying a model might not be nice, but also does not do much harm.

**Reply by mha1** [reply]
We are on the same page. Checking  the live status (rx connected) would help. If live no one touch resetting outputs, copying to subtrims, inserting deleting mix line without popup because this can lead to destroyed servos. Unfortunately the last stage of setting up a model has to be with the live model.

**Comment by offer-shmuely** [comment]
For destructive operations, I agree.
And it is true for some non touch operations as well (like the delete model) that pressed by mistake due to non sensitive operation on the wheel

**Comment by Eldenroot** [comment]
Yes pleasE!
